relayer/pkg/chainlink/keys/keystore.go

```diff
@@ -1,18 +1,54 @@
 package keys
 
 import (
+	"bytes"
 	"context"
+	"errors"
+	"fmt"
 	"math/big"
+	"sync"
 
+	"github.com/smartcontractkit/caigo"
 	caigotypes "github.com/smartcontractkit/caigo/types"
-
+	"github.com/smartcontractkit/chainlink-relay/pkg/loop"
 	"github.com/smartcontractkit/chainlink-relay/pkg/types"
 )
 
-//go:generate mockery --name Keystore --output ./mocks/ --case=underscore --filename keystore.go
+//go:generate mockery --name KeystoreAdapter --output ./mocks/ --case=underscore --filename keystore.go
+
+// KeystoreAdapter is a starknet-specific adaption layer to translate between the generic Loop Keystore (bytes) and
+// the type specific caigo Keystore (big.Int)
+// The loop.Keystore must be produce a byte representation that can be parsed by the caigo.Keystore implementation
+// Users of the interface are responsible to ensure this compatibility.
+type KeystoreAdapter interface {
+	caigo.Keystore
+	Loopp() loop.Keystore
+}
+
+// caigoAdapter implements [KeystoreAdapter]
+type caigoAdapter struct {
+	looppKs loop.Keystore
+}
+
+// Sign implements the caigo Keystore Sign func.
+func (ca *caigoAdapter) Sign(senderAddress string, hash *big.Int) (*big.Int, *big.Int, error) {
+	raw, err := ca.looppKs.Sign(context.Background(), senderAddress, hash.Bytes())
```

Comment:
noted in other PR - but having a context here would be useful so it doesn't block forever if something breaks :)

Reply:
done, thanks for catching